### PR TITLE
Add max_early_data_size to TicketEarlyDataInfo

### PR DESCRIPTION
**Pull request by FiloSottile**
Email to list coming.

### CODE REVIEW

**Comment by ekr** [comment]
@FiloSottile this needs a rebase.

**Comment by FiloSottile** [comment]
@ekr done, the damned GitHub editor doesn't detect conflicts. This is a bit ridiculous.

**Comment by ekr** [comment]
@FiloSottile Don't forget to email the list.

**Comment by FiloSottile** [comment]
https://www.ietf.org/mail-archive/web/tls/current/msg21486.html

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -3042,18 +3042,23 @@ This document defines one ticket extension, "ticket_early_data_info"
 ~~~~
    struct {
        uint32 ticket_age_add;
+       uint32 max_early_data_size;
    } TicketEarlyDataInfo;
 ~~~~
 
 This extension indicates that the ticket may be used to send 0-RTT data
-({{early-data-indication}})). It contains one value:
+({{early-data-indication}})). It contains two values:
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: (editorial nit: do we really need to have the number here?  Could it just be "Its contents are:"?)

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -3042,18 +3042,23 @@ This document defines one ticket extension, "ticket_early_data_info"
...
 
 ticket_age_add
 : A randomly generated 32-bit value that is used to obscure the age of the ticket that the
   client includes in the "early_data" extension.  The client-side ticket age is
   added to this value modulo 2^32 to obtain the value that is transmitted by
   the client.
+
+max_early_data_size
+: The maximum amount of 0-RTT data that the client is allowed to send when using
+  this ticket, in bytes. Only Application Data payload is counted.
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: A serer receiving more than this many bytes of 0-RTT data will abort the connection with a XYZ alert.

**Comment by ekr** [comment]
Merged by hand

**Pull request closed without merging**